### PR TITLE
[Feature][Permission] Reconstruction of permissions of resource center and monitoring center.

### LINKED ISSUE #10306: [Feature][Permission] Modify the permissions of project management, security center, data source center, data quality, Resource Center and Monitor Center module.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Modify the permissions of project management, security center, data source center, data quality, Resource Center and Monitor Center module.

### Use case

Unified management of resource permissions and function permissions extraction of public interfaces.

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by WangJPLeo**
## Purpose of the pull request

Abstract functions, resource permission verification, and reconstruction of project management, security center, data source center, data quality, resource center and monitoring center module permission management.

## Brief change log

close #10306 
 
## Verify this pull request

Manually verified the change by testing locally.

### CODE REVIEW

File: dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/UdfFuncMapper.xml [hunk 1]
```diff
@@ -62,11 +62,17 @@
...
-            and udf.id in (
-            select udf_id from t_ds_relation_udfs_user where user_id=#{userId}
-            union select id as udf_id from t_ds_udfs where user_id=#{userId})
+        <if test="ids != null and ids.size() > 0">
+            and udf.id in
+            <foreach collection="ids" item="i" open="(" close=")" separator=",">
+                #{i}
+            </foreach>
         </if>
+<!--        <if test="userId != 0">-->
```
Comment (lenboo) [inline dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/UdfFuncMapper.xml:71]: We can delete the unused code. Maybe you can fix it later

**Review by lenboo — APPROVED** [review]
+1

File: dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ResourceMapper.xml [hunk 2]
```diff
@@ -60,17 +60,12 @@
...
-            and (
-                    d.user_id=#{userId}
-                    <if test="resIds != null and resIds.size() > 0">
-                        or d.id in
-                        <foreach collection="resIds" item="i" open="(" close=")" separator=",">
-                            #{i}
-                        </foreach>
-                    </if>
-                )
+        where d.type=#{type} and d.pid=#{id} and u.id = #{userId}
```
Comment (user-GitHub-user) [inline dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ResourceMapper.xml:63]: why no need d.user_id = u.id ?

**Comment by Leijingwen** [comment]
image
image

Why is the code written disabled？

image
image

**Pull request merged**